logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt

```diff
@@ -151,4 +159,32 @@ class ConversationDataSource(
                 wrapApiRequest { clientApi.listClientsOfUsers(it) }.map { memberMapper.fromMapOfClientsResponseToRecipients(it) }
             }
     }
+
+    override suspend fun createOne2OneConversationWithTeamMate(otherUserId: UserId): Either<CoreFailure, ConversationId> = wrapApiRequest {
+        conversationApi.createNewConversation(
+            createConversationRequest = CreateConversationRequest(
+                qualifiedUsers = listOf(idMapper.toApiModel(otherUserId)),
+                access = emptyList(),
+                accessRole = listOf(),
+                convTeamInfo = null,
+                messageTimer = null,
+                receiptMode = 0,
+                conversationRole = "",
+                protocol = null,
+                name = ""
+            )
+        )
+    }.map { idMapper.fromApiModel(it.id) }
+
+    override suspend fun getOne2OneConversationByUserId(otherUserId: UserId): Either<StorageFailure, ConversationId?> {
+        return wrapStorageRequest {
+            observeConversationList()
+                .flatMapMerge { it.asFlow() }
+                .flatMapMerge { getConversationDetailsById(it.id) }
+                .filterIsInstance<ConversationDetails.OneOne>()
+                .firstOrNull { otherUserId == it.otherUser.id }
+                ?.conversation?.id
+        }
+    }
```

Comment:
This function name says it would return a whole Conversation, but it only returns the `ConversationId?`.

Perhaps we can return everything, and whoever is calling this function can use the ID or whatever else they need.
```suggestion
    override suspend fun getOne2OneConversationDetailsByUserId(otherUserId: UserId): Either<StorageFailure, ConversationDetails.OneOne?> {
        return wrapStorageRequest {
            observeConversationList()
                .flatMapMerge { it.asFlow() }
                .flatMapMerge { getConversationDetailsById(it.id) }
                .filterIsInstance<ConversationDetails.OneOne>()
                .firstOrNull { otherUserId == it.otherUser.id }
        }
    }
```

Reply:
Could you add some tests for this function?